llmfoundry/command_utils/train.py

```diff
@@ -518,8 +517,11 @@ def train(cfg: DictConfig) -> Trainer:
     # TP config
     if tp_config is not None:
         strategy = tp_config.pop('strategy', None)
-        assert isinstance(strategy, str), '`strategy` must be in `tp_config`.'
-        tp_config['layer_plan'] = build_tp_strategies(strategy, model)
+        layer_plan = build_tp_strategies(strategy, model)
+        tp_config = {
```

Comment:
the compose tp config has other keys allowed. should we just be constructing that dataclass here?

Reply:
+1

Reply:
done